src/java/org/apache/cassandra/index/sai/disk/v1/sortedterms/SortedTermsWriter.java

```diff
@@ -90,30 +80,29 @@ public class SortedTermsWriter implements Closeable
      *
      * @param componentName the component name for the {@link SortedTermsMeta}
      * @param metadataWriter the {@link MetadataWriter} for storing the {@link SortedTermsMeta}
-     * @param termsData where to write the prefix-compressed terms data
+     * @param termsOutput where to write the prefix-compressed terms data
      * @param termsDataBlockOffsets  where to write the offsets of each block of terms data
-     * @param trieOutput where to write the trie that maps the terms to point ids
+     * @param blockShift the block shift that is used to determine the block size
      */
     public SortedTermsWriter(String componentName,
                              MetadataWriter metadataWriter,
-                             IndexOutput termsData,
+                             IndexOutput termsOutput,
                              NumericValuesWriter termsDataBlockOffsets,
-                             IndexOutputWriter trieOutput) throws IOException
+                             int blockShift) throws IOException
     {
         this.componentName = componentName;
         this.metadataWriter = metadataWriter;
-        this.trieOutput = trieOutput;
-        SAICodecUtils.writeHeader(this.trieOutput);
-        this.trieWriter = new IncrementalDeepTrieWriterPageAware<>(trieSerializer, trieOutput.asSequentialWriter());
-        SAICodecUtils.writeHeader(termsData);
-        this.termsOutput = termsData;
-        this.bytesStartFP = termsData.getFilePointer();
+        SAICodecUtils.writeHeader(termsOutput);
+        this.blockShift = blockShift;
+        this.blockMask = (1 << this.blockShift) - 1;
+        this.termsOutput = termsOutput;
+        this.termsOutput.writeVInt(blockShift);
+        this.bytesStartFP = termsOutput.getFilePointer();
         this.offsetsWriter = termsDataBlockOffsets;
     }
 
     /**
      * Appends a term at the end of the sequence.
-     * Terms must be added in lexicographic order.
```

Comment:
Are they still added in unsigned byte-comparable order?

Reply:
No, I have changed this.